cmake/configure/configure_50_gmsh.cmake

```diff
@@ -17,5 +17,8 @@
 # Configuration for the gmsh executable:
 #
 
+macro(feature_gmsh_configure_external)
+  set(DEAL_II_GMSH_WITH_API ${GMSH_WITH_API})
+endmacro()
```

Comment:
@tjhei Here is the change that ensures that `DEAL_II_GMSH_WITH_API` only ever gets set if we enable gmsh support.